edi/models/edi_exchange_record_mixin.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2020 ACSONE SA
+# Copyright 2020 Creu Blanca
+# @author Simone Orsi <simahawk@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class EDIExchangeRecordMixin(models.AbstractModel):
+    """Record that might have related EDI Exchange records
+    """
+
+    _name = "edi.exchange.record.mixin"
+    _description = "Abstract record where exchange records can be assigned"
+
+    exchange_record_ids = fields.One2many(
+        "edi.exchange.record",
+        inverse_name="res_id",
+        domain=lambda r: [("model", "=", r._name)],
+    )
+    exchange_record_count = fields.Integer(compute="_compute_exchange_record_count")
+    missing_exchange_records = fields.Boolean(
+        compute="_compute_has_missing_exchange_records"
+    )
+
+    def _get_exchange_record_configuration(self):
```

Comment:
Is not clear what this method does/should return.
As per the schema of returned data I'd say it should always be a dict as it makes always clear what you get.
Let's talk about this maybe?

Reply:
I added more information there, I hope this is easier now.